webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/smartling/SmartlingResultProcessor.java

```diff
@@ -0,0 +1,105 @@
+package com.box.l10n.mojito.service.thirdparty.smartling;
+
+import com.amazonaws.SdkClientException;
+import com.box.l10n.mojito.service.blobstorage.Retention;
+import com.box.l10n.mojito.service.blobstorage.s3.S3BlobStorage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+@ConditionalOnProperty(value = "l10n.ThirdPartyTMS.impl", havingValue = "ThirdPartyTMSSmartling")
+@Component
+public class SmartlingResultProcessor {
+
+    static Logger logger = LoggerFactory.getLogger(SmartlingResultProcessor.class);
+
+    private static final String NAME_FORMAT = "%s_%s_%s.zip";
+
+    @Autowired
```

Comment:
`@Autowired(required=false)`

but to make it more generic you should use `StructuredBlobStorage`, it will use whatever blob storage is configured and specify a prefix (that how we avoid collision at a system level)

Reply:
I've used `S3BlobStorage` considering this is meant to be uploaded to S3, not through the generic interface.